applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -400,8 +400,9 @@ namespace Kratos
         
         for ( unsigned int point_number = 0; point_number < integration_points.size(); point_number++ )
         {   
-            const double detJ0 = CalculateDerivativesOnReferenceConfiguration(J0, InvJ0, DN_DX, point_number, integration_method);
-            const double IntegrationWeight = GetIntegrationWeight(integration_points, point_number, detJ0) * thickness;
+            // NOTE: The integration of the mass matrix is always performed in the initial configuration (Ted Belytschko book 2014 page 91)
```

Comment:
ok, however you'll have to reimplement it for UL if you want to be able to remesh

Reply:
Why, because the initial coordinates will be lost?

Reply:
Just read the comment... If you remedy this assumption is not valid (it is where UL and TL are no longer identical)